Source/JavaScriptCore/runtime/DateConstructor.cpp

```diff
@@ -71,6 +71,37 @@ static double millisecondsFromComponents(JSGlobalObject* globalObject, const Arg
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
 
+    auto toIntegerOrInfinity = [](double d) {
+        return trunc(std::isnan(d) ? 0.0 : d + 0.0);
+    };
+
+    // https://tc39.es/ecma262/#sec-maketime
+    auto makeTime = [](double hour, double min, double sec, double ms) {
+        return ((hour * msPerHour + min * msPerMinute) + sec * msPerSecond) + ms;
+    };
+
+    // https://tc39.es/ecma262/#sec-makeday
+    auto makeDay = [](double year, double month, double date) {
+        double additionalYears = std::floor(month / 12);
+        double ym = year + additionalYears;
+        if (!std::isfinite(ym))
+            return PNaN;
+        double mm = month - additionalYears * 12;
```

Comment:
`adjustedMonth` or `monthWithinYear`

Reply:
Ditto. This is aligned to the spec linked in the comment.